sources/rest_api/client.py

```diff
@@ -94,39 +118,37 @@ def request(
         return self._send_request(prepared_request)
 
     def get(
-        self, path: str = "", params: Optional[Dict[str, Any]] = None, **kwargs: Any
+        self, path: str, params: Optional[Dict[str, Any]] = None, **kwargs: Any
     ) -> Response:
         return self.request(path, method="get", params=params, **kwargs)
 
     def post(
-        self, path: str = "", json: Optional[Dict[str, Any]] = None, **kwargs: Any
+        self, path: str, json: Optional[Dict[str, Any]] = None, **kwargs: Any
     ) -> Response:
         return self.request(path, method="post", json=json, **kwargs)
 
     def paginate(
         self,
-        path: str = "",
-        method: Literal["get", "post"] = "get",
+        path: str,
+        method: str = "get",
         params: Optional[Dict[str, Any]] = None,
         json: Optional[Dict[str, Any]] = None,
-        auth: Optional[AuthBase] = None,
+        auth: Optional[AuthConfigBase] = None,
         paginator: Optional[BasePaginator] = None,
-        data_selector: Optional[Union[str, List[str]]] = None,
+        data_selector: Optional[jsonpath.TJsonPath] = None,
         response_actions: Optional[List[Dict[str, Any]]] = None,
         hooks: Optional[Dict[str, Any]] = None,
-    ) -> Generator[Any, None, None]:
+    ) -> Iterator[PageData[Any]]:
```

Comment:
I believe the correct type is Generator, not Iterator. Even though it's allowed to type it as an iterator:

```
>>> def infinite_stream(start: int):
...     while True:
...         yield start
...         start += 1
...
>>> type(infinite_stream(1))
<class 'generator'>
```